src/lib/Form/Data/Role/RoleAssignmentCreateData.php

```diff
@@ -29,35 +27,21 @@ class RoleAssignmentCreateData implements TranslationContainerInterface
 
     /**
      * @var \Ibexa\Contracts\Core\Repository\Values\Content\Section[]
-     *
-     * @Assert\Expression(
-     *     "this.getLimitationType() != 'section' or (this.getLimitationType() == 'section' and value != [])",
-     *     message="validator.define_subtree_or_section_limitation"
-     * )
      */
+    #[Assert\Expression("this.getLimitationType() != 'section' or (this.getLimitationType() == 'section' and value != [])", message: 'validator.define_subtree_or_section_limitation')]
```

Comment:
This looks less readable.

Reply:
Well, Rector did what it could, we need to adjust manually.

Reply:
I know, just pointing that out.

Reply:
I expect us to only re-adjust the formatting. Previous one with multiple lines was alright.

As for your concerns for expression language syntax in context of Symfony Constraints, according to Symfony docs, `value` can be used when attribute is attached to a property.

> The [expression](https://symfony.com/doc/current/reference/constraints/Expression.html#reference-constraint-expression-option) option is the expression that must return true in order for validation to pass. Learn more about the [expression language syntax](https://symfony.com/doc/current/reference/formats/expression_language.html).

As for test coverage... Who knows...

Reply:
Adjusting the formatting to previous version is enough from my perspective.

Reply:
> As for test coverage... Who knows...

That concerned me deeply. I've checked - there was 0 coverage for validation errors for that form, neither for unit/integration test setup nor behat.

I've added some basic coverage for validation via attributes, so we can be sure it works. I'm not saying that we should cover all assertions, because we don't need to test what Symfony has already covered. But if an assertion contains custom code - expression or regex, this rather calls for a test case. Hence added relevant.

The test case showed that, as I expected, the expression could be simplified. Done so as well.